core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -65,6 +65,20 @@
 /*--------------------------------------------------------------------------*/
 /*--------------------------------------------------------------------------*/
 
+cudaStream_t Kokkos::Impl::cuda_get_deep_copy_stream() {
+  static cudaStream_t s = nullptr;
+  if (s == nullptr) {
+    cudaStreamCreate(&s);
+  }
+  return s;
+}
+
+Kokkos::Cuda const &
+Kokkos::Impl::cuda_get_execution_space_with_deep_copy_stream() {
+  static Cuda space(Kokkos::Cuda::DeepCopyTag{});
```

Comment:
why this instead of just 
```c++
static Cuda space(cuda_get_deep_copy_stream())
```

Reply:
That' what I had initially done in 6a9b3f4a2059243eda765c662e1454f132137ae5 but it gave https://github.com/kokkos/kokkos/pull/3856#issuecomment-801454324

But now that I look at it again, I agree that the current version is quite convoluted